### PR TITLE
universe: sort incoming transfer proofs by block height

### PR DESCRIPTION
**Pull request by Roasbeef**
In this commit, we attempt a simple sorting routine to address: https://github.com/lightninglabs/taproot-assets/issues/591.

Without these, we'll try to validate transfer proofs out of order, which is akin verifying an orphan transaction in the Bitcoin mempool. Rather than do a full on topological sort, we instead just sort by block height. As we don't support spending unconfirmed change, this should be enough to serialize the contents of a block.

### CODE REVIEW

File: universe/syncer.go [hunk 1]
```diff
@@ -210,13 +211,21 @@ func (s *SimpleSyncer) syncRoot(ctx context.Context, remoteRoot BaseRoot,
 	)
 
 	// We use an error group to simply the error handling of a goroutine.
+	// This goroutine will handle reading in batches of new leaves to
+	// insert into the DB. We'll fee the output of the goroutines below
```
Comment (guggero) [inline universe/syncer.go:215]: nit: s/fee/feed

File: universe/syncer.go [hunk 2]
```diff
@@ -243,22 +252,42 @@ func (s *SimpleSyncer) syncRoot(ctx context.Context, remoteRoot BaseRoot,
...
+		})
 	if err != nil {
 		return err
 	}
 
+	// If this is a tranfer tree, then we'll collect all the items as we
+	// need to sort them to ensure we can validate them in dep order.
+	if !isIssuanceTree {
+		transferLeaves := fn.Collect(transferLeafProofs)
+		sort.Slice(transferLeaves, func(i, j int) bool {
```
Comment (guggero) [inline universe/syncer.go:283]: I think this'll work for now. Though I wonder: Will we ever want to make it possible to have subsequent transfers within the same block? Assuming an interactive scheme where participants would accept proofs as being valid before they have the block header populated (e.g. in a pocket universe construction)?

**Review by guggero — APPROVED** [review]
LGTM :tada: 

An integration test would've been nice. But I'll add one when addressing https://github.com/lightninglabs/taproot-assets/issues/575 or https://github.com/lightninglabs/taproot-assets/issues/571 anyway, so not a blocker.

**Review by jharveyb — APPROVED** [review]
Looks good - added some other detail to #591.

File: universe/syncer.go (hunk 2 above)
Comment (Roasbeef) [reply universe/syncer.go:283]: Yeah I think so, then we would use something like an `scid` here, that takes into account: tx index, and output index as well.

Comment (dstadulis) [reply universe/syncer.go:283]: Will this sort be run again if there's a block reorg?

**Comment by Roasbeef** [comment]
Superseded by #594

**Pull request closed without merging**